### PR TITLE
Add data limitation

### PR DESCRIPTION
**Pull request by gkumbhat**
### Description
- Add training data limit which should be configurable by module and model

### CODE REVIEW

File: caikit_nlp/modules/text_generation/peft_prompt_tuning.py [hunk 1]
```diff
@@ -366,6 +367,16 @@ def train(
             verbalizer,
         )
 
+        # Check if data is within limit allowed for this module and model
+        max_num_examples = get_config().training_data_limit.get(cls.MODULE_ID, {}).get(base_model_name, -1)
+
+        if max_num_examples > 0:
+            error.value_check(
```
Comment (evaline-ju) [inline caikit_nlp/modules/text_generation/peft_prompt_tuning.py:374]: maybe good to have a tiny unit test case with temp config? :)

File: caikit_nlp/config/config.yml [hunk 2]
```diff
@@ -30,5 +30,9 @@ unload_tgis_prompt_artifacts: false
 master_addr: localhost
 master_port: 29550
 
+training_data_limit:
+  6655831b-960a-4dc5-8df4-867026e2cd41:
```
[the remark this answers is not in the document]
Comment (gkumbhat) [reply caikit_nlp/config/config.yml:34]: good point. Will add that

File: caikit_nlp/modules/text_generation/peft_prompt_tuning.py [hunk 3]
```diff
@@ -366,6 +367,16 @@ def train(
...
+                "<NLP77627434E>",
+                len(train_stream) <= max_num_examples,
```
[the remark this answers is not in the document]
Comment (gkumbhat) [reply caikit_nlp/modules/text_generation/peft_prompt_tuning.py:380]: yep, thats definitely required

**Review by evaline-ju — APPROVED** [review]
LGTM but we should probably not merge until caikit bump

**Pull request merged**